### ISSUE TITLE
Accessibility Colors

### ISSUE THREAD

**Issue by KianBadie** [KianBadie/cell-atlas#18]
Recoloring certain elements of the site to pass accessibility test.

**Comment by KianBadie** [comment]
@coiko The colors are unfortunately proving to be a little trickier than initially expected. To start off with the buttons, they do not pass the accessibility standards of having a 3:1 contrast ratio with the dark gray background. It looks like the options (while staying in the "blue area" we initially had) is to make the buttons lighter with less color or lighter with a strong blue (screenshots below).

If you wanted to keep the dark blue, then we would have to increase the background of the text section, but I personally don't think that looks that good in relation to the other elements on the page (3rd image). It's possible, but it would require finding new colors for the other elements shown in the picture as well, as they all kind of blend together.

What are your thoughts?

## Screenshots
![Screenshot from 2020-12-14 16-17-37](https://user-images.githubusercontent.com/18221058/102151630-1bbe4380-3e28-11eb-8430-9da2ac7eb094.png)
![Screenshot from 2020-12-14 16-17-59](https://user-images.githubusercontent.com/18221058/102151631-1cef7080-3e28-11eb-87fa-6426b2b1a26a.png)
## Making background brighter
![Screenshot from 2020-12-14 16-23-03](https://user-images.githubusercontent.com/18221058/102151913-cc2c4780-3e28-11eb-833f-9059d4ea891d.png)

**Comment by coiko** [comment]
@KianBadie Sorry this is a pain!  It seems like it's always the "simplest" things that are the trickiest.  I like the lighter-with-less color option (the second screenshot).  That might also be a good option for the links that appear on the dark-grey background (in the text).  I like that better than changing the to a lighter background, since that makes the bulk of the text harder to read.  I'm also not sure how critical the low contrast is for the buttons themselves, since the text on them stands out.  What do you think?

**Comment by KianBadie** [comment]
@coiko Our section pages actually have a good score from Google's accessibility automated test after I changed a couple small things (around 96 / 100) before the colors are even changed. I still imagine it is important for all the accessibility standards to be met (especially within academics), but I just wanted to put into perspective Where we are at in terms of our score from Google. In addition, it is an automated test, so it won't catch everything.

And yes they are! That is exactly what I was thinking with the lighter color working for the links. That is a good question with the text. I just did some further research and webAIM.org has these resources (pictures below). It seems like our case meets the scenario in the first bullet of the first image. The second image is an example.

## Screenshots
![Screenshot from 2020-12-14 16-48-10](https://user-images.githubusercontent.com/18221058/102153571-60e47480-3e2c-11eb-9cc4-bed54efe8973.png)
![Screenshot from 2020-12-14 16-48-46](https://user-images.githubusercontent.com/18221058/102153573-617d0b00-3e2c-11eb-8d3c-2485c97447a2.png)

**Comment by coiko** [comment]
Thanks @KianBadie!  That's a great explanation.  Then let's go with the lighter/less-color option!

**Comment by KianBadie** [comment]
@coiko No Problem! I will also modify some of the other colors Google is pointing out and post pictures!

**Comment by KianBadie** [comment]
@coiko I attached a screenshot of a version with no contrast complaints from Google (except the CalTech text color in the footer, but through my fiddling it seems like it's not possible to keep a orange-ish color with a grey background). The colors in the image make the site look pretty different. If anything, this shot at the accessibility colors could be just a starting point to get the brain jogging on possible shades of blue to keep the original feel of the site. My apologies for doing this right before our agreed deadline. I did not think these adjustments would be this tricky.

After typing the above, I did a little reading on opinions of the WCAG (the accessibility guidelines I have been trying to meet with these colors). I saw torn opinions on treating the guidelines as the standard vs using them as, well, guidelines (as in false negatives can result from sticking to the standards). For those on the latter side, they brought up the importance of user testing to find true preferences. But it seemed like those on the former side believed that wouldn't account for who the guidelines are actually trying to help. Quite the pickle. 

It seems like this is the last visual aspect of the site to confirm in order to have things in a shareable state so that others can perform user experience tests. I think all other major features are implemented (will do a check tomorrow to make sure of that). I'm not sure if this should be decided before others test, or after testing. Maybe with some sleep I will have new thoughts, but I am interested in hearing yours.

## Screenshot
![Screenshot from 2020-12-14 22-27-55](https://user-images.githubusercontent.com/18221058/102179684-68bc0d00-3e5c-11eb-9246-7aed2d38bec4.png)

**Comment by coiko** [comment]
@KianBadie Thank you so much for looking into this!  I'm sorry I wasn't more careful about it in the designs so you have to grapple with it now.  And yes, I completely agree that we should meet the guidelines wherever we can - they're there for a very good reason.  I do see what you mean about the different site feel, though.  Looking at it again fresh this morning, I wonder if you wouldn't mind going back to something you showed earlier, which is a lighter background color so the buttons and citation text stand out.  Is it possible to use that much lighter background just for the bottom section of the text box?  We could also lighten the top a bit, just not so much that it would make the text (which is the main thing people will be spending a lot of time reading) too hard to see.  Then the buttons and links in the bottom section could be the original dark blue, and the links in the main text could be the color you have them now, which looks nice.  What do you think?

**Comment by KianBadie** [comment]
@coiko No worries at all, really! My apologies as well. If I investigated this issue a lot earlier, it would have at least got the conversation going and it would have been something we could tweak while we were adding features. 

I wouldn't mind trying anything out! It would be possible to make it lighter just for the bottom section, I'll fiddle with that and post some screenshots. I think lightening the video citation text would require darkening the white text that currently exist in there, but it would definitely provide more contrast with the existing blue text. And yes I agree with what you said about lightening to top. I'll try both versions (lightening just the bottom and lightening the whole text section) and post both versions. I think what you shared is a good idea and I will post an update soon!

**Comment by KianBadie** [comment]
@coiko Alright, I have a promising update. I think the results from the fiddling didn't turn out too bad. The overall text section becomes in the clear in terms of accessibility starting at **2a** (except for the in-text link colors). I think lightening the background text too much will cause trouble with the link colors. I also showed how bright we can go in **3b** while staying in the standards for the white text (the background would need to change of course).

While looking at these, I'm thinking if we lighten things just a bit, maybe we could darken the background in **3a**? I can post a picture of that too, but I wanted to post something to get some feedback before going further. What are you thoughts?

I personally kind of like the citation bar in **2a**. It makes me feel like I'm looking at one of the F1 cars.

## 1. Increasing learn more to minimum brightness to pass standards
![part1-learnmore-change](https://user-images.githubusercontent.com/18221058/102258112-6c827a80-3ec2-11eb-9986-9b67bc69d761.png)
## 2a. Increrasing citation to minimum brightness to pass standards (while also darkening white text to minimum darkness to pass)
![part2-videocitation-change](https://user-images.githubusercontent.com/18221058/102258144-76a47900-3ec2-11eb-9a32-0fa9e7333d15.png)
## 2b. Version where I darkened blue citation text to same shade as buttons so citation bar can be darker and "Collected by/doi" text a little lighter than previous version
![part2-version2](https://user-images.githubusercontent.com/18221058/102258269-9b005580-3ec2-11eb-97db-f80f0fcff6ce.png)
## 3a. Slightly lighter text background
![part3-slightlylighter-background](https://user-images.githubusercontent.com/18221058/102258497-d7cc4c80-3ec2-11eb-9d0c-c355cf70ec06.png)
## 3b. As light as you can go with the text background (would have to darken overall background)
![part3-aslightasyoucango](https://user-images.githubusercontent.com/18221058/102258563-e74b9580-3ec2-11eb-8615-6a10b07bb4b2.png)

**Comment by coiko** [comment]
@KianBadie I love 2a!  I think it's perfect!  All we need to do is lighten the link color in the main text, right?  Thanks so much for all your work on this.  I'm sorry it was a pain but I think it really improved it!

**Comment by KianBadie** [comment]
@coiko Awesome! And for the text section yes, there are some other things I am going to check (the text in the nav menu was throwing problems, but that should be much easier than this since its just black/grey on white text). I will provide another update very soon after I do some more fiddling. And as always, not a problem at all; I'm glad it improved!

**Comment by KianBadie** [comment]
@coiko I added underlines to the links on the page to fulfill another requirement where if links aren't underlined by default they need to have a certain contrast ratio with the surrounding text. I assumed adding the underline would look better but let me know if you want to experiment with the alternative. Out of the 3 options for link colors on this darker background (which would apply to section/appendix pages), which one do you think is best?

## 1.
![Screenshot from 2020-12-15 12-32-04](https://user-images.githubusercontent.com/18221058/102269614-bf176300-3ed1-11eb-8578-b62c3702865b.png)
## 2.
![Screenshot from 2020-12-15 12-32-13](https://user-images.githubusercontent.com/18221058/102269616-bfaff980-3ed1-11eb-91f0-ee010a65c2b5.png)
## 3.
![Screenshot from 2020-12-15 12-32-42](https://user-images.githubusercontent.com/18221058/102269618-c0489000-3ed1-11eb-9d16-e67e44ea796c.png)

**Comment by coiko** [comment]
Thanks @KianBadie!  I think I like option 3 the best.  Also, the underlining looks great!

**Comment by KianBadie** [comment]
@coiko I changed the look of the appendix pages to use similar colors that we added to the section pages. Thoughts on this? The differences are the link text and the "+" buttons.

![Screenshot from 2020-12-15 14-44-27](https://user-images.githubusercontent.com/18221058/102281627-250ce600-3ee4-11eb-90b6-061f03e3d8d0.png)

**Comment by coiko** [comment]
@KianBadie I think it looks fantastic!

**Comment by KianBadie** [comment]
@coiko Great! The only thing that I added since the last picture is that all the links have underlines since it is next to non link text with a lower contrast.

**Comment by KianBadie** [comment]
@coiko Another page changed. I made the reference links in the chapter landing pages white and underlined. It didn't seem like any flavor of blue was going to mesh with the grey background. It seems like grey backgrounds only work with white or black text. What do you think of this? 

![Screenshot from 2020-12-15 15-08-32](https://user-images.githubusercontent.com/18221058/102283588-94380980-3ee7-11eb-84e6-a51db0df99f1.png)

**Comment by coiko** [comment]
@KianBadie It's perfect!

**Comment by KianBadie** [comment]
@coiko A subtle update, but before the current page was not only italicized but also the text was lightened a little bit in the nav menu. I got rid of that to meet the accessibility standard. So now it is just italicized.   

![Screenshot from 2020-12-15 16-04-11](https://user-images.githubusercontent.com/18221058/102287408-3efff600-3eef-11eb-9901-1211ab74194a.png)

**Comment by coiko** [comment]
@KianBadie I love it!  (I had noticed that was hard to read.)  Thanks so much for taking care of all this!

**Comment by KianBadie** [comment]
@coiko No problem!

**Comment by KianBadie** [comment]
@coiko Going through the rest of the site, it looks like we might be ok on the color contrast. The only complaint Google gives is on the orange CalTech logo in the footer. Unfortunately, because of the grey background it does not seem like any sort of orange will work out. Would it still be satisfying if that text was white? It is also what the CalTech library does for their CalTech logo on a grey footer at the botttom of their [home page](https://www.library.caltech.edu/)

The only other thing that I think I might have to tweak is the landing page. It has an image for a background, white text, and the dark blue button/header. I'll post pictures for that, but thankfully I think that will be a lot less finicky/tricky than the section pages (unless the "simplest things are always the trickiest" rule strikes again, but I doubt it with the landing page). The different components are different shades (light text/dark button) so I think there's a chance I will be increasing the lightness on that button.

In total, I think this color contrast thing will be wrapped up tonight!

**Comment by coiko** [comment]
@KianBadie That's great news!  Yes, for now let's go with the white text for the Caltech logo.  In the future, maybe we can fix it by adding a white outline around the orange to keep the color pop but still have it readable?

**Comment by KianBadie** [comment]
@coiko Awesome! That makes things simpler. Yeah that could work out, I did see something in the guidelines mentioning how text outlines can change things but I didn't read more into it. That could possibly work out in the future!

**Comment by KianBadie** [comment]
@coiko Well, it looks like all the colors are good! The only thing that can be tweaked is the "Read now" button on the landing page. It is difficult to find what technically meets the guidelines because the background is an image, so I have to do it manually. I tried measuring a random grey spot around the button and treated that as the background color and increased the button brightness to have the correct amount of contrast against that. I also decreased the brightness of the background image to be safe. What do you think of this? 

Thankfully, this is the last contrast thing to deal with. So tweaking it should be less difficult than figuring out the section pages.

![Screenshot from 2020-12-15 20-21-09](https://user-images.githubusercontent.com/18221058/102304992-bba4cb80-3f13-11eb-932c-cff850c8c489.png)

**Comment by coiko** [comment]
@KianBadie Wonderful!  Thank you so much.  I like the extra space at the top of the page now, too!

**Comment by KianBadie** [comment]
@coiko Awesome, glad you like it!
